src/phpDocumentor/Reflection/Php/Factory/ClassConstantIterator.php

```diff
@@ -0,0 +1,135 @@
+<?php
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @copyright 2010-2015 Mike van Riel<mike@phpdoc.org>
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      http://phpdoc.org
+ */
+
+
+namespace phpDocumentor\Reflection\Php\Factory;
+
+use PhpParser\Comment\Doc;
+use PhpParser\Node\Stmt\ClassConst;
+
+/**
+ * This class acts like a combination of a ClassConst and Const_ to be able to create constant descriptors using a normal strategy.
+ */
+final class ClassConstantIterator implements \Iterator
```

Comment:
Missing a `getValue` method; constants have methods that should be interpreted using the PrettyPrinter and returned